plugins/org.jkiss.dbeaver.data.transfer/src/org/jkiss/dbeaver/tools/transfer/DataTransferSettings.java

```diff
@@ -179,7 +180,17 @@ private void initializePipes(
             // Both producers and consumers specified
             // Processor belongs to non-database nodes anyway
             if (initProducers.length != initConsumers.length) {
-                throw new IllegalArgumentException("Producers number must match consumers number");
+                // Something went wrong
+                if (initProducers.length < initConsumers.length && initProducers[0] instanceof DatabaseTransferProducer) {
```

Comment:
Is it possible, by any chance, to have an array of producers (`initProducers`) that has zero elements?